### PR TITLE
Add more documentation about common beam convolution

### PR DESCRIPTION
**Pull request by keflavich**
Relatively minor docs addition.

@e-koch might need your help reminding me where I need doctest-skip directives etc.

Also I hope this is all syntactically correct.... did it from memory w/o testing 😬

### CODE REVIEW

File: docs/commonbeam.rst [hunk 1]
```diff
@@ -21,6 +21,49 @@ will have a very small effect on the size of the common beam.
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The typical use case for calculating a common beam is to convolve one or more
+data sets to a common resolution.  Using radio-beam with spectral-cube, you
+would first compute the common beam, then convolve all data sets to that beam.
+
+For a varying-resolution spectral cube, one in which there are different beams
+for each channel, the process looks like:
+
+.. python::
+
```
Comment (e-koch) [inline docs/commonbeam.rst:35]: add a spectral-cube import

File: docs/commonbeam.rst [hunk 2]
```diff
@@ -21,6 +21,49 @@ will have a very small effect on the size of the common beam.
...
+Convolution to a common resolution
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The typical use case for calculating a common beam is to convolve one or more
+data sets to a common resolution.  Using radio-beam with spectral-cube, you
+would first compute the common beam, then convolve all data sets to that beam.
+
+For a varying-resolution spectral cube, one in which there are different beams
+for each channel, the process looks like:
+
+.. python::
```
Comment (e-koch) [inline docs/commonbeam.rst:33]: the docs build fails here. I can't remember why this is, but the rest of our docs don't have this block. They just start with the `>>>`

File: docs/commonbeam.rst [hunk 3]
```diff
@@ -21,6 +21,49 @@ will have a very small effect on the size of the common beam.
...
+   >>> common_beam = cube.beams.common_beam()
+   >>> cb_cube = cube.convolve_to(common_beam)
+
+If you have two different data sets, you would follow a similar process:
+
+.. python::
+
+   >>> cube1 = SpectralCube.read('cube1.image')
+   >>> cube2 = SpectralCube.read('cube2.image')
+   >>> common_beam = Beams([cube1.beam, cube2.beam]).common_beam()
```
Comment (e-koch) [inline docs/commonbeam.rst:47]: Can simplify from https://github.com/radio-astro-tools/radio-beam/pull/102:

```suggestion
   >>> common_beam = cube1.beam.commonbeam_with(cube2.beam)
```
Which doesn't look simplified... but at least it doesn't require making a new `Beams` object. Also may set the record for most number of `beam`s in a single line.

File: docs/commonbeam.rst [hunk 4]
```diff
@@ -21,6 +21,49 @@ will have a very small effect on the size of the common beam.
...
+
+
+Note that this process is equivalent to calculating the common beam,
+deconvolving the original data's beam, and convolving with the resulting kernel.
+
+.. python::
+
+   >>> cube1 = SpectralCube.read('cube1.image')
+   >>> cube2 = SpectralCube.read('cube2.image')
+   >>> common_beam = Beams([cube1.beam, cube2.beam]).common_beam()
```
Comment (e-koch) [inline docs/commonbeam.rst:60]:
```suggestion
   >>> common_beam = cube1.beam.commonbeam_with(cube2.beam)
```

**Review by e-koch — CHANGES_REQUESTED** [review]
We may need need `  # doctest: +SKIP` everywhere. I don't think we have test data cubes kept in radio-beam. Unless we can grab one via a utility function added in spectral-cube,

File: docs/commonbeam.rst (hunk 1 above)
Comment (keflavich) [reply docs/commonbeam.rst:35]: yeah we need to skip these tests; the cubes don't exist

**Comment by e-koch** [comment]
Not sure either.

**Comment by keflavich** [comment]
apparently I'm bad at intersphinx too; I want to reference the spectral-cube docs on smoothing but that failed:
https://github.com/radio-astro-tools/radio-beam/actions/runs/3337835531/jobs/5524692842#step:5:121
`/home/runner/work/radio-beam/radio-beam/docs/commonbeam.rst:58:unknown document: 'spectral-cube:smoothing'`

**Review by e-koch — APPROVED** [review]

**Pull request merged**